community/modules/scripts/wait-for-startup/main.tf

```diff
@@ -13,13 +13,27 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-data "google_compute_instance" "vm_instance" {
+
+resource "null_resource" "validate_instance_names" {
+  lifecycle {
+    precondition {
+      condition     = var.instance_name != null || length(var.instance_names) > 0
+      error_message = "At least one instance name must be provided"
+    }
+  }
+}
+
+data "google_compute_instance" "vm_instance_single" {
+  count = var.instance_name == null ? 0 : 1
```

Comment:
It exhibits the same problem. I only see two options, either:
* let it fail;
* deprecate `var.instance_name`.

Reply:
I'd like to avoid making a change that could break currently working blueprints.

I think this change would be big improvement to usability of the module.  Given that I haven't been able to repro the failure with any valid HPC Toolkit blueprint (but it is possible to make it fail with hand-written Terraform code) I would be in favor of just letting it fail.

@nick-stroud, any thoughts?

Reply:
I can repro with the following snippet:
```
  - id: new-bucket
    source: community/modules/file-system/cloud-storage-bucket
    settings:
      random_suffix: true

  - source: ./modules/compute/vm-instance
    id: workstation
    use:
    - network1
    settings:
      name_prefix: $(new-bucket.gcs_bucket_name)
      machine_type: n2-standard-2
      instance_count: 2
      startup_script: |
        sleep 10

  - id: wait0
    source: ./community/modules/scripts/wait-for-startup
    settings:
      instance_name: $(workstation.name[0])
```

There is no random naming behavior in vm-instance, but we name it after the bucket that has a random name.

Interestingly `instance_names: $(workstation.name)` has no issue, as the length of array is known even though the names are not. I assume it would be possible to randomize that as well although, much less likely to occur, I think.

One option would be to have the singular form always run (no count) and just make the script tolerant to null value being passed (exit immediately). In that case I think you could even go back to the original formulation where you combine the singular with the array, just don't perform the set function.